### ISSUE TITLE
I only get test ads even when I set isTesting to false

### ISSUE THREAD

**Issue by tyshawngaines** [ratson/cordova-plugin-admob-free#309]
I need help, when I switched from admob-simple i had no problems receiving ads but now that i am using admob free, when i show my ads they always return as test ads.my app has been on the market for months and this is the first time i had this issue of only test ads showing. anyone know how to set this to false in order to display real ads?

document.addEventListener('deviceready', function(){
function adInit(){
var admobid = {}
if (/(android)/i.test(navigator.userAgent)) {
  admobid = {
	  interstitial:'ca-app-pub-xxxxxxxx/xxxxxxxx',
	  rewardvideo:'ca-app-pub-xxxxxxxx/xxxxxxxxx'
  };
	
	//interstitial
	admob.interstitial.config({
    id: admobid.interstitial,
    isTesting: false,
    autoShow: false,
  	});
			
	//reward video	
	admob.rewardvideo.config({
    id: admobid.rewardvideo,
    isTesting: false,
    autoShow: false,})	
}	
	adInit()
	try{
	admob.rewardvideo.prepare()	
	admob.interstitial.prepare()	
	}
	catch(ad){}
}})

**Comment by diegoestacho999** [comment]
Have mediations been set up? From what I saw in the plugin's documentation here, you need to set up mediations for ads to show. At least that's what I did to receive the reward videos.

**Comment by moomdate** [comment]
+1 same issue

**Comment by P48L0** [comment]
any uppdate on this? i also dont see any ads in producion mode, but i do see the placeholder when isTesting is true...

**Comment by aboutwd** [comment]
Yeah, I can't get LIVE ads to show in production mode either.    Been trying for 2 weeks.

- "Test Ads" work fine in the android emulator.
-  LIVE ads work normal when previewing in the ionic DevApp - using "ionic serve -c" 
-  But once the app is built and installed (both signed and unsigned) on either the emulator or a device, I get nothing.

But I can't get any of the Ionic Admob plugins to work properly. 
I would appreciate any suggestions as well.  Thanks.

ionic 4.5.0
cordova 8.1.2

**Comment by tyshawngaines** [comment]
I gave up months ago and went with admob pro. Nearly same implementation

On Dec 11, 2018 7:32 PM, "Rob Goodwin" <notifications@github.com> wrote:

Yeah, I can't get LIVE ads to show in production mode either. Been trying
for 2 weeks.

   - "Test Ads" work fine in the android emulator.
   - LIVE ads work normal when previewing in the ionic DevApp - using
   "ionic serve -c"
   - But once the app is built and installed (both signed and unsigned) on
   either the emulator or a device, I get nothing.

But I can't get any of the Ionic Admob plugins to work properly.
I would appreciate any suggestions as well. Thanks.

ionic 4.5.0
cordova 8.1.2

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/ratson/cordova-plugin-admob-free/issues/309#issuecomment-446416198>,
or mute the thread

**Comment by KyadaKrupali** [comment]
i get some error  Please help me ASAP

HomePage.html:65 ERROR TypeError: Object(...) is not a function
    at AdMobFreeBanner.config (index.js:91)
    at HomePage.webpackJsonp.199.HomePage.showBannerAd (home.ts:44)
    at Object.eval [as handleEvent] (HomePage.html:65)
    at handleEvent (core.js:13589)
    at callWithDebugContext (core.js:15098)
    at Object.debugHandleEvent [as handleEvent] (core.js:14685)
    at dispatchEvent (core.js:10004)
    at core.js:10629
    at HTMLElement.<anonymous> (platform-browser.js:2628)
    at t.invokeTask (polyfills.js:3)

**Comment by mohannadt1** [comment]
You can’t see real ads directly. It’s normal that are not be showed.

Since February 2019, google added a new rule to avoid developer from using signed apk’s on virtual machines under vpn that were clicking to earn directly with virtual clients.

 

So just publish your app and wait about 2 or 3 days for the review, then real ads will appear.

**Comment by omar300** [comment]
> You can’t see real ads directly. It’s normal that are not be showed.
> 
> Since February 2019, google added a new rule to avoid developer from using signed apk’s on virtual machines under vpn that were clicking to earn directly with virtual clients.
> 
> So just publish your app and wait about 2 or 3 days for the review, then real ads will appear.

is there a way i can test this? 
i just added the ad to code, 
for now im testing and its showing the test ad but if i change isTesting value to false nothing is showing, i atleast want to see how it will look like. 
can keep trying for a couple of days and see if the ad shows in couple days?
this is unsigned apk.

**Comment by mohannadt1** [comment]
> > لا يمكنك رؤية إعلانات حقيقية مباشرة. من الطبيعي ألا تظهر.
> > منذ شباط (فبراير) 2019 ، أضافت google قاعدة جديدة لتجنب المطور من استخدام ملفات apk موقعة على أجهزة افتراضية تحت vpn تم النقر عليها لكسب مباشرة مع العملاء الظاهري.
> > لذلك فقط انشر تطبيقك وانتظر حوالي يومين أو ثلاثة أيام للمراجعة ، ثم ستظهر إعلانات حقيقية.
> 
> هل هناك طريقة يمكنني اختبار هذا؟
> لقد قمت للتو بإضافة الإعلان إلى الشفرة ،
> في الوقت الحالي لاختبار im وعرض إعلان الاختبار ، لكن إذا قمت بتغيير قيمة الاختبار إلى شيء خاطئ لا يظهر ، فأنا أريد أن أرى كيف سيبدو.
> يمكن أن تستمر في المحاولة لبضعة أيام ومعرفة ما إذا كان الإعلان يظهر في يومين؟
> هذا غير موقَّع apk.

I will talk to you about personal experience
If you previously published the application
  1- Try to change the application package name with a test ad code
You will see a test ad
2- Then rename the old one and upload it to the store
